### ISSUE TITLE
INFLUXD_CONFIG_PATH does not work if there is a `.` in the directory name

### ISSUE THREAD

**Issue by jmmk** [influxdata/influxdb#20696]
According to the docs: https://docs.influxdata.com/influxdb/v2.0/reference/config-options/#influxdb-configuration-file
> On startup, influxd will check for a config.* in the INFLUXD_CONFIG_PATH directory.

__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Create `config.toml` in a directory, for example `/vol/mounted-drive/config.toml`
2. Set `INFLUXD_CONFIG_PATH` to the parent directory of the config file, e.g. `INFLUXD_CONFIG_PATH=/vol/mounted-drive`
3. Run `influxd`

__Expected behavior:__
`influxd` should find and use `/vol/mounted-drive/config.toml`

__Actual behavior:__
The custom configuration is not used

__Environment info:__

* System info: `Linux 4.4.0-1085-aws x86_64`
* InfluxDB version: `InfluxDB 2.0.3 (git: fe04d346df) build_date: 2020-12-15T01:00:16Z`

---

This may not be a bug, but rather an inaccuracy in the documentation. The workaround is to simply point `INFLUXD_CONFIG_PATH` at `config.toml` itself, rather than the parent directory.

**Comment by danxmoran** [comment]
@jmmk thanks for the report! I'm not able to reproduce locally. Are you able to `stat /vol/mounted-drive/config.toml` as the user running `influxd`?

We use [viper](https://github.com/spf13/viper) to implement our config-loading, including the search logic that seems to be broken for you. The code there makes an `os.Stat` call for each possible `config.{ext}` under `INFLUXD_CONFIG_PATH`, so I wonder if we can reproduce an error from outside the program.

**Comment by jmmk** [comment]
If I'm not mistaken, the behavior you are describing is found here: https://github.com/influxdata/influxdb/blob/master/kit/cli/viper.go#L63

```
	if configPath := v.GetString("CONFIG_PATH"); configPath != "" {
		switch strings.ToLower(path.Ext(configPath)) {
		case ".json", ".toml", ".yaml", ".yml":
			v.SetConfigFile(configPath)
		case "":
			v.AddConfigPath(configPath)
		}
	} else {
		// defaults to looking in same directory as program running for
		// a file with base `config` and extensions .json|.toml|.yaml|.yml
		v.SetConfigName("config")
		v.AddConfigPath(".")
	}
```

I think I see the issue here. The documentation for `path.Ext` states:

> Ext returns the file name extension used by path. The extension is the suffix beginning at the final dot in the final slash-separated element of path; it is empty if there is no dot. 

I didn't accurately describe the setup in my original comment - my directory actually contained a `.` like `/vol/.directory`. `path.Ext` returns `.directory` for this which does not match either of the switch cases and therefore does not add any paths for viper to load configuration from.

So the documentation is correct, but there does seem to be a bug after all that directories with `.` in them are not handled properly.

**Comment by danxmoran** [comment]
@jmmk thanks for the debugging assist, I think you're absolutely right. It should (hopefully) be a quick fix

**Comment by danxmoran** [comment]
Fix will be included in next week's 2.0.4 release
